### PR TITLE
LPS-69102 Replace props usages with new method to get the maximum upload servlet request content length

### PR DESCRIPTION
**Pull request by pei-jung**
Hey Roberto,

Can you please take a look at the last commit (1ce15ad) and see if my changes are ok?

This pull is a prerequisite of [LPS-69102](https://issues.liferay.com/browse/LPS-69102) where the Upload Servlet Request portal properties will be migrated to OSGi configuration.

Thanks!

/cc @jorgeferrer @drewbrokke

### CODE REVIEW

File: modules/apps/collaboration/wiki/wiki-editor-configuration/src/test/java/com/liferay/wiki/editor/configuration/internal/WikiAttachmentImageHTMLEditorConfigContributorTest.java [hunk 1]
```diff
@@ -165,6 +170,11 @@ public void testItemSelectorURLWhenAllowBrowseAndValidWikiPage()
 		wikiAttachmentImageHTMLEditorConfigContributor.setItemSelector(
 			_itemSelector);
 
+		suppress(
```
Comment (robertoDiaz) [inline modules/apps/collaboration/wiki/wiki-editor-configuration/src/test/java/com/liferay/wiki/editor/configuration/internal/WikiAttachmentImageHTMLEditorConfigContributorTest.java]: Why is this needed??

At first sight this change doesn't seem to be needed...

Why don't you need to suppress other similar methods??

Thanks for the info

Comment (pei-jung) [inline modules/apps/collaboration/wiki/wiki-editor-configuration/src/test/java/com/liferay/wiki/editor/configuration/internal/WikiAttachmentImageHTMLEditorConfigContributorTest.java]: That's what I thought at first sight too, but the test failed (https://github.com/pei-jung/liferay-portal/pull/187#issuecomment-297830908). When I traced down the error, it turns out that it's trying to call the new method. 

in the test, we call `wikiAttachmentImageHTMLEditorConfigContributor.populateConfigJSONObject`, which calls `getItemSelectorURL` and triggers `getUploadItemSelectorCriterion` that tries to create a new instance of `UploadItemSelectorCriterion`. Since I replaced the props usage with the new method in 1d5c095aeb5713de7e475cca46d2a00ce5e4c931, the test complains that it couldn't find the new method. And looking at the test, it seems it doesn't depend on `getUploadItemSelectorCriterion`, so I suppress it.

**Comment by pei-jung** [comment]
Hi @robertoDiaz,

Thank you for taking the time to review my changes. Can you explain what you mean by 
> we agreed that it could be much better to create and consume an abstraction to avoid couple the portlets with an implementation?

And if possible, can you show me an example that does this? Thanks!

File: modules/apps/collaboration/wiki/wiki-editor-configuration/src/test/java/com/liferay/wiki/editor/configuration/internal/WikiAttachmentImageHTMLEditorConfigContributorTest.java (hunk 1 above)
Comment (robertoDiaz) [inline modules/apps/collaboration/wiki/wiki-editor-configuration/src/test/java/com/liferay/wiki/editor/configuration/internal/WikiAttachmentImageHTMLEditorConfigContributorTest.java]: It doesn't seem to be right, because we are modifiying the behaviour of the logic we are testing.

It seems to be a dependency error (can you try to upgrade the kernel version in the test bundle and recheck??)

**Comment by robertoDiaz** [comment]
With this changes we are going to use a static method located in an implementation class:

UploadServletRequestImpl --> implementation class
public static long getMaxSize() --> Static method

Doing this we are going to introduce a direct dependency between all the clients of the method (a bunch of bundles) and the implementation class.

It could be better to move this method to the interface --> UploadServletRequest and set it as a default method, so the dependency will not be tied to the impl but to the abstraction.

**Comment by pei-jung** [comment]
Hey @robertoDiaz,

Thanks for the explanation! That makes sense. I need to look more into this and find a different way to approach this. Thanks again for taking the time to review this. I'll close the pr for now.

**Comment by robertoDiaz** [comment]
Thanks to you @pei-jung :)
